tests/integration/test_syscollector/test_syscollector_configuration.py

```diff
@@ -0,0 +1,434 @@
+        - The `case_test_syscollector_deactivation.yaml` file provides the test cases.
+    '''
+    file_monitor = FileMonitor(LOG_FILE_PATH) if sys.platform == 'win32' else None
+    evm.check_syscollector_is_disabled(file_monitor=file_monitor, timeout=T_10)
+
+
+@pytest.mark.parametrize('configuration, metadata', zip(t2_configurations, t2_config_metadata), ids=t2_case_ids)
+def test_syscollector_all_scans_disabled(configuration, metadata, set_wazuh_configuration,
+                                         configure_local_internal_options_module, truncate_monitored_files,
+                                         daemons_handler_function):
+    '''
+    description: Check that each scan is disabled.
+
+    test_phases:
+        - Configure syscollector.
+        - Configure modulesd in debug mode.
+        - Truncate the log.
+        - Restart analysisd, wazuh-db and modulesd
+
+    wazuh_min_version: 4.4.0
+
+    tier: 0
+
+    parameters:
+        - configuration:
+            type: dict
+            brief: Configuration loaded from the template located in `configuration` folder.
+        - metadata:
+            type: dict
+            brief: Test case metadata.
+        - set_wazuh_configuration:
+            type: fixture
+            brief: Set wazuh configuration using the configuration template.
+        - configure_local_internal_options_module:
+            type: fixture
+            brief: Configure the local internal options file.
+        - truncate_monitored_files:
+            type: fixture
+            brief: Truncate the log file before and after the test execution.
+        - daemons_handler_function:
+            type: fixture
+            brief: Handler of Wazuh daemons for each test case.
+
+    assertions:
+        - Check if a specific scan is disabled and not triggered.
+
+    input_description:
+        - The `configuration_syscollector_scans_disabled.yaml` file provides the module configuration for this test.
+        - The `case_test_all_scans_disabled.yaml` file provides the test cases.
+    '''
+    file_monitor = None
+    check_functions = [evm.check_hardware_scan_started, evm.check_os_scan_started, evm.check_network_scan_started,
+                       evm.check_packages_scan_started, evm.check_ports_scan_started, evm.check_processes_scan_started]
+    if sys.platform == 'win32':
+        check_functions.append(evm.check_hotfixes_scan_started)
+
+    # Check that no scan is triggered
+    for check_f in check_functions:
+        # Expected: the function must throw a TimoutError
+        with pytest.raises(TimeoutError):
+            # Overwrite the default timeout (because the test configuration)
+            file_monitor = FileMonitor(LOG_FILE_PATH)
+            check_f(file_monitor=file_monitor, timeout=T_10)
+            pytest.fail(f"It seems that a scan was triggered. This check has a match in the log: {check_f.__name__}")
```

Comment:
This won't work, because if `update position` is True, the logs can appear while you are checking the others, so it will never detect that cases.

You have to rework this and move these checks to differences use cases, by parametrizing them

Reply:
Added a comment in dce76b38b